android/forms-screenlet-demo/app/src/main/java/com/liferay/mobile/formsscreenletdemo/presenter/HomePresenter.java

```diff
@@ -0,0 +1,112 @@
+package com.liferay.mobile.formsscreenletdemo.presenter;
+
+import com.liferay.apio.consumer.model.Thing;
+import com.liferay.mobile.android.service.Session;
+import com.liferay.mobile.formsscreenletdemo.R;
+import com.liferay.mobile.formsscreenletdemo.contract.HomeViewContract;
+import com.liferay.mobile.formsscreenletdemo.service.APIOFetchResourceService;
+import com.liferay.mobile.formsscreenletdemo.util.Constants;
+import com.liferay.mobile.formsscreenletdemo.util.DemoUtil;
+import com.liferay.mobile.formsscreenletdemo.util.ResourceType;
+import com.liferay.mobile.formsscreenletdemo.view.HomeActivity;
+import com.liferay.mobile.push.Push;
+import com.liferay.mobile.screens.context.SessionContext;
+import com.liferay.mobile.screens.context.storage.CredentialsStorageBuilder;
+import com.liferay.mobile.screens.ddm.form.model.FormInstanceRecord;
+import com.liferay.mobile.screens.ddm.form.service.APIOFetchLatestDraftService;
+import com.liferay.mobile.screens.thingscreenlet.screens.views.Custom;
+import com.liferay.mobile.screens.util.LiferayLogger;
+import kotlin.Unit;
+import org.json.JSONException;
+
+/**
+ * @author Luísa Lima
+ */
+public class HomePresenter implements HomeViewContract.HomePresenter {
+
+	private HomeActivity homeActivity;
+
+	@Override
+	public void onActivityCreated(HomeActivity homeActivity) {
+		this.homeActivity = homeActivity;
+
+		setupForPushNotification();
+	}
+
+	@Override
+	public void onActivityDestroyed() {
```

Comment:
Here you could use a BasePresenter to abstract this part of the implementation for every presenter.
`this.homeActivity = homeActivity;`
Your BasePresenter should have an activity property, and if you need to add more behaviors to your onActivityCreated you just override and call super before.